src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -0,0 +1,522 @@
+              shortName = "sgaPath",
+              fullName  = "fullPathToSGA",
+              optional  = false)
+    public String pathToSGA = null;
+
+    @Argument(doc       = "An URI to headerless file where each line contains the absolute path to one raw FASTQ file " +
+                            "corresponding to a putative breakpoint",
+              shortName = "fl",
+              fullName  = "fastqList",
+              optional  = false)
+    public String pathToFASTQListFile = null;
+
+    @Argument(doc       = "A path to a directory to write results to.",
+              shortName = "outDir",
+              fullName  = "outputDirectory",
+              optional  = false)
+    public String outputDir = null;
+
+    @Argument(doc       = "Number of threads to use when running sga.",
+              shortName = "t",
+              fullName  = "threads",
+              optional  = true)
+    public int threads = 1;
+
+    @Argument(doc       = "To run k-mer based read correction, filter and duplication removal in SGA or not, with default parameters.",
+              shortName = "correct",
+              fullName  = "correctNFilter",
+              optional  = true)
+    public boolean runCorrectionSteps = false;
+
+    @Override
+    public boolean requiresReads(){
+        return true;
+    }
+
+    @Override
+    public void runTool(final JavaSparkContext ctx){
+
+        final JavaRDD<String> rawFASTQFiles = ctx.textFile(Paths.get(pathToFASTQListFile).toAbsolutePath().toString());
+        final JavaPairRDD<Long, URI> seqsArrangedByBreakpoints = rawFASTQFiles.mapToPair(RunSGAViaProcessBuilderOnSpark::assignFASTQToBreakpoints);
+
+        final Path sgaPath = Paths.get(pathToSGA);
+
+        // map paired reads to FASTA file (temp files live in temp dir that cleans self up automatically) then feed to SGA for assembly
+        final JavaPairRDD<Long, SGAAssemblyResult> assembledContigs = seqsArrangedByBreakpoints.mapToPair(entry -> performAssembly(entry, sgaPath, threads, runCorrectionSteps));
+
+        validateAndSaveResults(assembledContigs, outputDir);
+    }
+
+    /**
+     * Converts from a line of text, delimited with a tab, where the first entry is the breakpoint ID and the second entry
+     *   is URI to its associated FASTQ file on HDFS.
+     * @param recordLine    a line of text containing the breakpoint ID and URI to associated FASTQ file
+     * @return              a pair constructed by splitting the string into the ID part and the URI part
+     */
+    @VisibleForTesting
+    static Tuple2<Long, URI> assignFASTQToBreakpoints(final String recordLine){
+        final String[] recordForOneBreakPoint = recordLine.split("\t");
+        final Long breakpointID = Long.valueOf(recordForOneBreakPoint[0]);
+        try{
+            final URI fastqURI = new URI(recordForOneBreakPoint[1]);
+
+            return new Tuple2<>(breakpointID, fastqURI);
+        }catch (final Exception e){
```

Comment:
What Exception are you expecting here? Better to catch a subtype of exception if possible. If it's an expected exception type, you can handle it as below, but some for other types of exception that might happen you'd still want to throw the exception up I bet.